lib/internal/crypto/diffiehellman.js

```diff
@@ -118,7 +118,7 @@ function DiffieHellman(sizeOrKey, keyEncoding, generator, genEncoding) {
   if (typeof sizeOrKey !== 'number')
     sizeOrKey = toBuf(sizeOrKey, keyEncoding);
 
-  if (!generator) {
```

Comment:
I'm still not sure what should be the behaviour of this when generator is function

Reply:
I don't think this is correct, I think you need to do this instead:
```diff
diff --git a/lib/internal/crypto/diffiehellman.js b/lib/internal/crypto/diffiehellman.js
index e2106b211ba..19116c97bf8 100644
--- a/lib/internal/crypto/diffiehellman.js
+++ b/lib/internal/crypto/diffiehellman.js
@@ -122,9 +122,9 @@ function DiffieHellman(sizeOrKey, keyEncoding, generator, genEncoding) {
     generator = DH_GENERATOR;
   } else if (typeof generator === 'number') {
     validateInt32(generator, 'generator');
-  } else if (generator !== true) {
+  } else if (typeof generator === 'string') {
     generator = toBuf(generator, genEncoding);
-  } else {
+  } else if(!isArrayBufferView(generator) && !isAnyArrayBuffer(generator)) {
     throw new ERR_INVALID_ARG_TYPE(
       'generator',
       ['number', 'string', 'ArrayBuffer', 'Buffer', 'TypedArray', 'DataView'],
```